File: oauth2library/src/main/java/ca/mimic/oauth2library/OAuthError.java
package ca.mimic.oauth2library;

public class OAuthError {
    protected String error;
    protected String error_description;
    protected String error_uri;

    protected transient Exception exception;

    public OAuthError(Exception e) {
        exception = e;
    }

    public String getError() {
        return error;
    }

    public String getErrorDescription() {
        return error_description;
    }

    public String getErrorUri() {
        return error_uri;
    }

    public Exception getErrorException() {
        return exception;
    }
}

File: oauth2library/src/main/java/com/pillohealth/oauth2library/OAuthError.kt
package com.pillohealth.oauth2library

class OAuthError(e: Exception) {
    var error: String? = null
        private set
    var errorDescription: String? = null
        private set
    var errorUri: String? = null
        private set

    @Transient
    var errorException: Exception
        private set

    init {
        errorException = e
    }
}
